File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
 import java.util.Arrays;

class Solution {

    public String solution(String[] participant, String[] completion) {
        String answer = "";
            
//1번째 방법 
//sorting,loof를 활용한 솔루션
//1. participant, completion 두 배열을 정렬-sort-한다 (그러면 알파벳순으로 abc, ab 이렇게 리스트화 됨!!)
        Arrays.sort(participant);
        Arrays.sort(completion);
//2. 기준을 completion으로 두고 두 배열이 다를 때까지 없는 선수를 찾는다.
        int i = 0;
        for(; i<completion.length ; i++) {
            if(!participant[i].equals(completion[i]))
                break;
        }
//3. 예외처리 - 혹여나 마지막까지 왔다면, 마지막 선수가 완주하지 못한 선수다. (자동적으로)
        return participant[i];
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    hash_dict={} # hash dictionary
    sumHash=0
    for i in participant:
        hash_dict[hash(i)]=i # participant hash 인덱스 넣기
        sumHash+=hash(i) # hash 총합
        
    for j in completion:
        sumHash-=hash(j) # completion의 hash값들 빼기
        
    return hash_dict[sumHash] # 남은 hash값에 해당하는 선수가 완주 못한 선수